### PR TITLE
feat(core-flows,medusa,types,utils): add rules to promotion endpoints + workflow

### PR DESCRIPTION
**Pull request by riqwan**
what:

- adds endpoint + workflow to add rules to a promotion
  - promotion rules
  - application target rules
  - application buy rules
- updates promotion module:
  - add*Rules to return created PromotionRules instead of Promotion
  - remove*Rules return void similar to other remove methods

### CODE REVIEW

File: packages/core-flows/src/promotion/workflows/add-rules-to-promotions.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { CreatePromotionRuleDTO, PromotionRuleTypes } from "@medusajs/types"
+import { WorkflowData, createWorkflow } from "@medusajs/workflows-sdk"
+import { addRulesToPromotionsStep } from "../steps"
+
+type WorkflowInput = {
```
Comment (olivermrbl) [inline packages/core-flows/src/promotion/workflows/add-rules-to-promotions.ts]: todo: Move to `@medusajs/types`

File: packages/core-flows/src/promotion/steps/add-rules-to-promotions.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      (await promotionModule.removePromotionBuyRules(
+        id,
+        buyRuleIds.map((id) => ({ id }))
+      ))
+
+    targetRuleIds.length &&
+      (await promotionModule.removePromotionBuyRules(
+        id,
+        targetRuleIds.map((id) => ({ id }))
+      ))
```
Comment (olivermrbl) [inline packages/core-flows/src/promotion/steps/add-rules-to-promotions.ts:74]: question: Can we parallelize?

File: packages/core-flows/src/promotion/steps/add-rules-to-promotions.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+    const createdPromotionBuyRules =
+      ruleType === RuleType.BUY_RULES
+        ? await promotionModule.addPromotionBuyRules(data.id, data.rules)
+        : []
+
+    const createdPromotionTargetRules =
+      ruleType === RuleType.TARGET_RULES
+        ? await promotionModule.addPromotionTargetRules(data.id, data.rules)
+        : []
```
Comment (olivermrbl) [inline packages/core-flows/src/promotion/steps/add-rules-to-promotions.ts:32]: question: Can we parallelize?

**Review by olivermrbl — APPROVED** [review]
LGTM, great work. Minor comments. 

Can't figure out if it makes sense to squash the endpoints, but I think we'll find out when building out the admin UI for it.

**Comment by riqwan** [comment]
@olivermrbl Same, I had an urge to do `promotions/[id]/[rule_type]` as well 😄

File: packages/core-flows/src/promotion/steps/add-rules-to-promotions.ts (hunk 2 above)
Comment (riqwan) [reply packages/core-flows/src/promotion/steps/add-rules-to-promotions.ts:74]: Only one of these would ever be run on a workflow run, parallelising doesn't help much here.

Comment (olivermrbl) [reply packages/core-flows/src/promotion/steps/add-rules-to-promotions.ts:74]: Ah yes, don't know how I missed that. Discard the comment.

**Pull request merged**